modules/openapi-generator/src/main/resources/dart2/api.mustache

```diff
@@ -1,129 +1,131 @@
+    Object postBody{{#bodyParam}} = {{{paramName}}}{{/bodyParam}};
 
     // verify required params are set
     {{#allParams}}
       {{#required}}
-    if({{paramName}} == null) {
-     throw ApiException(400, "Missing required param: {{paramName}}");
+    if ({{{paramName}}} == null) {
+     throw ApiException(400, "Missing required param: {{{paramName}}}");
     }
       {{/required}}
     {{/allParams}}
 
     // create path and map variables
-    String path = "{{{path}}}".replaceAll("{format}","json"){{#pathParams}}.replaceAll("{" + "{{baseName}}" + "}", {{{paramName}}}.toString()){{/pathParams}};
+    String path = "{{{path}}}".replaceAll("{format}","json"){{#pathParams}}.replaceAll("{" + "{{{baseName}}}" + "}", {{{paramName}}}.toString()){{/pathParams}};
 
     // query params
     List<QueryParam> queryParams = [];
     Map<String, String> headerParams = {};
     Map<String, String> formParams = {};
     {{#queryParams}}
     {{^required}}
-    if({{paramName}} != null) {
+    if ({{{paramName}}} != null) {
     {{/required}}
-      queryParams.addAll(_convertParametersForCollectionFormat("{{collectionFormat}}", "{{baseName}}", {{paramName}}));
+      queryParams.addAll(_convertParametersForCollectionFormat("{{{collectionFormat}}}", "{{{baseName}}}", {{{paramName}}}));
     {{^required}}
     }
     {{/required}}
     {{/queryParams}}
     {{#headerParams}}
-    headerParams["{{baseName}}"] = {{paramName}};
+    headerParams["{{{baseName}}}"] = {{{paramName}}};
     {{/headerParams}}
 
     List<String> contentTypes = [{{#consumes}}"{{{mediaType}}}"{{#hasMore}},{{/hasMore}}{{/consumes}}];
 
     String nullableContentType = contentTypes.isNotEmpty ? contentTypes[0] : null;
-    List<String> authNames = [{{#authMethods}}"{{name}}"{{#hasMore}}, {{/hasMore}}{{/authMethods}}];
+    List<String> authNames = [{{#authMethods}}"{{{name}}}"{{#hasMore}}, {{/hasMore}}{{/authMethods}}];
 
-    if(nullableContentType != null && nullableContentType.startsWith("multipart/form-data")) {
+    if (nullableContentType != null && nullableContentType.startsWith("multipart/form-data")) {
       bool hasFields = false;
       MultipartRequest mp = MultipartRequest(null, null);
       {{#formParams}}
       {{^isFile}}
-      if ({{paramName}} != null) {
+      if ({{{paramName}}} != null) {
         hasFields = true;
-        mp.fields['{{baseName}}'] = parameterToString({{paramName}});
+        mp.fields["{{{baseName}}}"] = parameterToString({{{paramName}}});
       }
       {{/isFile}}
       {{#isFile}}
-      if ({{paramName}} != null) {
+      if ({{{paramName}}} != null) {
         hasFields = true;
-        mp.fields['{{baseName}}'] = {{paramName}}.field;
-        mp.files.add({{paramName}});
+        mp.fields["{{{baseName}}}"] = {{{paramName}}}.field;
```

Comment:
That's actually not a best-practice :) However, if given a choice either to have consistency or to conform, then I choose consistency. So the reasoning behind using double-quotes over single-quotes is because the Dart generator wraps strings with double-quotes.

So I would rather see double-quotes in the generated code all over the place, instead of having inconsistency.

Makes sense?

Reply:
@agilob 

I have looked into DartClientCodegen.java and I can see where the const and " are being injected. I would rather have single quotes to be honest, so I think I will go ahead and roll back the changes for double-quotes.

I also saw that the same Java code injects `const []` and `const {}` (in `toDefaultValue(Schema)`), which is not backward compatible with previous versions of the same file. What do you reckon, should we keep the const in the Java file or put it in the Mustache template?